plugins/node/instrumentation-undici/src/undici.ts

```diff
@@ -139,12 +140,35 @@ export class UndiciInstrumentation extends InstrumentationBase<UndiciInstrumenta
     diagnosticChannel: string,
     onMessage: ListenerRecord['onMessage']
   ) {
-    const channel = diagch.channel(diagnosticChannel);
-    channel.subscribe(onMessage);
+    // NOTE: later versions of `@types/node` include proper type definitions
+    // for diagnostics channel. However updating it breaks compilation :(
```

Comment:
Should there be a separate issue to follow-up on that? Or is there one already?

Reply:
I think is related to typescript because it spits typing errors from other modules. I guess we could ref to https://github.com/open-telemetry/opentelemetry-js/issues/4870